crates/bevy_input/src/button_input.rs

```diff
@@ -298,6 +287,22 @@ where
     pub fn get_just_released(&self) -> impl ExactSizeIterator<Item = &T> {
         self.just_released.iter()
     }
+
+    /// Returns the current state of the input.
+    pub fn pressed_state(&self, input: T) -> InputState {
+        if self.pressed(input) {
+            return InputState::Pressed;
+        }
+        InputState::Released
+    }
+}
+
+/// State of the input
+pub enum InputState {
```

Comment:
I would prefer to not introduce this in this PR. This is "just" an alternative API for bools. Having two ways to do the same way is confusing, and I personally think bools are the more ergonomic + user friendly approach in this context. Feel free to propose it in a separate PR though.